### ISSUE TITLE
Failed to build 0.14.8 on rustc 1.26.2 on alpine

### ISSUE THREAD

**Issue by sebasmagri** [dtolnay/syn#470]
When building on 1.26.2 on alpine 3.8, I get the following:

```
   Compiling syn v0.14.8
error[E0277]: the trait bound `proc_macro2::TokenStream: std::convert::From<proc_macro::TokenStream>` is not satisfied
   --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/syn-0.14.8/src/lib.rs:637:19
    |
637 |     parse2(tokens.into())
    |                   ^^^^ the trait `std::convert::From<proc_macro::TokenStream>` is not implemented for `proc_macro2::TokenStream`
    |
    = note: required because of the requirements on the impl of `std::convert::Into<proc_macro2::TokenStream>` for `proc_macro::TokenStream`

error[E0277]: the trait bound `proc_macro2::TokenStream: std::convert::From<proc_macro::TokenStream>` is not satisfied
   --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/syn-0.14.8/src/buffer.rs:228:27
    |
228 |         Self::new2(stream.into())
    |                           ^^^^ the trait `std::convert::From<proc_macro::TokenStream>` is not implemented for `proc_macro2::TokenStream`
    |
    = note: required because of the requirements on the impl of `std::convert::Into<proc_macro2::TokenStream>` for `proc_macro::TokenStream`

error[E0277]: the trait bound `proc_macro2::TokenStream: std::convert::From<proc_macro::TokenStream>` is not satisfied
   --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/syn-0.14.8/src/synom.rs:320:28
    |
320 |         self.parse2(tokens.into())
    |                            ^^^^ the trait `std::convert::From<proc_macro::TokenStream>` is not implemented for `proc_macro2::TokenStream`
    |
    = note: required because of the requirements on the impl of `std::convert::Into<proc_macro2::TokenStream>` for `proc_macro::TokenStream`

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0277`.
error: Could not compile `syn`.
```

Any pointers on how to fix it?

EDIT: added details about alpine, since I have not been able to reproduce it in other platforms

**Comment by sebasmagri** [comment]
Thanks!

**Comment by dtolnay** [comment]
Thanks for the report. I believe this was caused by a misunderstanding in https://github.com/alexcrichton/proc-macro2/issues/52 / https://github.com/alexcrichton/proc-macro2/commit/f3d1b2c7818be3fc8f856433092b91c6a8b8cbeb. I published proc-macro2 0.4.12 with a fix.
